### ISSUE TITLE
Switch from Python to JavaScript

### ISSUE THREAD

**Issue by mgspross** [jalawelo/math-problem-generator#6]
After some discussion, we've decided that the combination of JavaScript (for programming language) and [Node.js](https://nodejs.org/en/about/) (for runtime) will be a better fit for this project.

Although Python is a perfectly fine language and would've been a valid choice for this project, we realized that the project maintainers collectively have more familiarity and experience working with JavaScript (both back-end and front-end) over Python: thus, the project is better served by choosing a language that we (as a group) have more proficiency in.

That said, we already did write some early proof-of-concept Python code for this project: this code is currently in the [`wip/python-example`](../tree/wip/python-example) branch. A good first step would be to convert this code to JavaScript and get it running in Node.js.

If anyone wants to tackle this, feel free to take a stab at it and open a new PR with a JavaScript version of that code. If you have any improvements to the program design in general, that's all the better, but a roughly one-to-one translation from Python to JavaScript would also still be useful, since we can always improve upon the code and overall project design once we have something tangible to work with.

**Comment by franchet** [comment]
Okay @mgspross I think I'll give it a go. It might take some time but I'll see what I can do and then we can take it from there.

**Comment by Icarus-Dragon** [comment]
Awesome. I'm gonna learn something new. I will do my best and get up to speed. We can do this !!!!

**Comment by newoga** [comment]
@franchet @Icarus-Dragon That's great guys. Let us know if you run into any issues and would like some direction.

**Comment by mgspross** [comment]
@franchet, @Icarus-Dragon: We want to pause work on this and make sure issue #9 gets done first, and resume work on this issue after that's done. Please check issue #9 for details.

I'll unblock this once #9 is resolved.

**Comment by mgspross** [comment]
Unblocking this since a fix for #9 has been resolved and merged.

**Comment by DelanoBannister** [comment]
what just happened






Delanoxbannister11@gmail.com
<Delanoxbannister11@gmail.com>

**Comment by mgspross** [comment]
@DelanoBannister: Sorry. The meeting was disconnected. I just wanted to wrap up by saying if you or @franchet have questions as you work on things, you can comment under the issue you are working on.

If the questions are more about the processes, you can still reach out to @dgabbadon so we can coordinate on those topics if needed.

**Comment by DelanoBannister** [comment]
thank you alot mike










Delanoxbannister11@gmail.com
<Delanoxbannister11@gmail.com>

**Comment by mgspross** [comment]
@franchet, @DelanoBannister (/cc @newoga, @dgabbadon): 

I just want to quickly clarify the requirements for completing this issue and what's expected:

1. **Due Date**: I am setting a due date of **2019-11-07 (next Thursday)** for this to be completed to keep the project moving forward. I will be checking this issue throughout next week during my evening hours for any comments, and will expecting a pull request (PR). Read on for more details.
2. **End Goal**: The JavaScript version of the program should behave _identically_ to the Python version that is currently in the [`wip/python-example`](https://github.com/jalawelo/math-problem-generator/tree/wip/python-example) branch. We aren't expecting a different program or different functionality at this stage.
3. **File Layout**: Keep the same file layout as the Python version: use the same file names, replacing `.py` with `.js` (for example, the JavaScript version of `formatters.py` should be named `formatters.js`, and so on). This will make it easier to review and discuss.
4. **Code Review**: If you need code to be reviewed while you are still working on this issue, submit your work-in-progress in a pull request to _this_ repository (not to your own fork). We can provide feedback in that PR as needed to ensure we're going in the right direction. New work can be pushed to an existing pull request, so only one PR needs to be created for this issue, and new commits can be pushed up as work progresses on it.  
    
    The "final" push to the PR should have only `.js` files in it, as the `.py` files can be removed after the program is fully converted into JavaScript and it's confirmed to be working.
5. **Feedback**: If there are any questions/concerns related to any of the above, please post a comment to this issue so that we can address things as they come up. We want to keep communication here in GitHub as much as possible, and communication outside of GitHub is currently not efficient.

**Comment by mgspross** [comment]
Closing this, since this requirement is now part of the general implementation requirements listed in issue #18.
